File: 01jupyter/nb/source/cs01/include/omp_parallel_rank.c
#include <stdio.h>
#include <omp.h>

int main() {
  printf("hello\n");
#pragma omp parallel
  {
    int omp_nthreads = omp_get_num_threads();
    int omp_rank = omp_get_thread_num();
    printf("world %3d/%3d\n", omp_rank, omp_nthreads);
  }
  printf("good bye\n");
  return 0;
}

File: 01jupyter/nb/source/02_parallel/include/omp_parallel_rank.f90
program omp_parallel_rank
  use omp_lib
  integer :: omp_nthreads, omp_rank
  print "(a)", "hello"
  !$omp parallel private(omp_nthreads, omp_rank)
  omp_nthreads = omp_get_num_threads()
  omp_rank = omp_get_thread_num()
  print "(a,i3,a,i3)", "world ", omp_rank, "/", omp_nthreads
  !$omp end parallel
  print "(a)", "good bye"
end program omp_parallel_rank
